201-managed-application-with-customprovider/azuredeploy.json

```diff
@@ -0,0 +1,87 @@
+{
+    "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "location": {
+            "type": "string",
+            "allowedValues": [ "australiaeast", "eastus", "westeurope" ],
+            "metadata": {
+                "description": "Location for the resources."
+            }
+        },
+        "appname": {
+            "type": "string",
+            "metadata": {
+                "description": "Managed app name"
+            }
+        },
+        "_artifactsLocation": {
+            "type": "string",
+            "metadata": {
+                "description": "The base URI where artifacts required by this template are located. When the template is deployed using the accompanying scripts, a private location in the subscription will be used and this value will be automatically generated."
+            },
+            "defaultValue": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/201-managed-application-with-customprovider/"
+        },
+        "_artifactsLocationSasToken": {
+            "type": "securestring",
+            "metadata": {
+                "description": "The sasToken required to access _artifactsLocation.  When the template is deployed using the accompanying scripts, a sasToken will be automatically generated."
+            },
+            "defaultValue": ""
+        }
+    },
+    "variables": {
+        "locklevel" : "None",
+        "description" : "Sample Managed application definition with custom resources and actions",
+        "displayName" : "Sample Managed application definition with custom resources",
+        "managedappdefname": "[concat(parameters('appname'), '_ApplicationDefinition')]",
+        "packageFileUri":"[uri(parameters('_artifactsLocation'), concat('artifacts/ManagedAppZip/pkg.zip', parameters('_artifactsLocationSasToken')))]",
+        "managedResourceGroupId" : "[concat(subscription().id,'/resourceGroups/', concat(parameters('appname'),'_managed'))]",
```

Comment:
remove concat and use subscriptionResourceId()

Reply:
That does not work either cause that function needs resource type. I tried doing : 
"managedResourceGroupId": "[subscriptionResourceId('resourceGroups', concat(parameters('applicationName'),'_managed'))]",

But that fails with error : 

 _Unable to evaluate template language function 'subscriptionResourceId': function requires exactly one multi-segmented argument which must be resource type including resource
provider namespace_